### PR TITLE
[OBSDEF-6043] federation add atlas store

### PR DESCRIPTION
**Pull request by bst525**
## Purpose
[OBSDEF-6043](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-6043)
Adds the atlas store to federation charts
This chart change can happen without any new federation service.  However, the next release of federation service will depend on the federation atlas client.  
Updating the chart first in the dependency chain.
It is using the same atlas definitions as the iam service.
**When merged**, federation will have same requirements as iamsvc for **single node deployments**.
` --set federation.atlas.replicaCount=1 `
If multiple replicas are deployed on single node, they will fail to start (unless this setting is added)
`  --set objectscale-iam.atlas.disableAntiAffinity=true`

Automation has been updated with ` --set federation.atlas.replicaCount=1 ` for single node installs

## PR checklist
- [ ] make test passed
- [x] make build passed
- [x] helm install <chart> passed
- [ ] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed

## Testing
https://asd-ecs-jenkins.isus.emc.com/jenkins/view/All/job/fedsvc-custom-ci/56/  PASSED
https://asd-ecs-jenkins.isus.emc.com/jenkins/job/charts-custom-ci/108/  PASSED
https://asd-ecs-jenkins.isus.emc.com/jenkins/job/iamsvc-custom-ci/148/ PASSED

https://asd-ecs-jenkins.isus.emc.com/jenkins/job/charts-custom-ci/109/  PASSED

Helm test is still failing.
```
helm unittest ecs-cluster objectscale-manager mongoose zookeeper-operator atlas-operator decks kahm dks-testapp fio-test sonobuoy dellemc-license service-pod helm-controller objectscale-graphql objectscale-vsphere objectscale-portal objectscale-gateway objectscale-iam pravega-operator bookkeeper-operator supportassist decks-support-store statefuldaemonset-operator influxdb-operator federation logging-injector dcm
Error: fork/exec /root/.local/share/helm/plugins/helm-unittest/untt: no such file or directory
Makefile:67: recipe for target 'test' failed
make: *** [test] Error 1
```

### CODE REVIEW

File: federation/templates/atlas_v1beta1_fedcluster.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+apiVersion: atlas.dellemc.com/v1beta1
+kind: AtlasCluster
+metadata:
+  name: {{ include "fedsvc.name" . }}-atlas
+spec:
+  replicas: {{ .Values.atlas.replicaCount }}
```
Comment (eddiepavkovic) [inline federation/templates/atlas_v1beta1_fedcluster.yaml:7]: missing `imagePullSecrets` required to be able to pull docker images from private registries: https://github.com/EMCECS/charts/blob/6943ed6856ed463408545312174416a538977b94/dcm/templates/atlas_v1beta1_dcmcluster.yaml#L9-L11

File: federation/values.yaml [hunk 2]
```diff
@@ -20,3 +20,22 @@ fedsvc:
...
+
+  # Disables the affinity rules for the Atlas Cluster, allowing it to run
+  # multiple replicas on a single node
+  disableAntiAffinity: false
+
+  # Configure the container image
+  image:
+    pullPolicy: IfNotPresent
+    repository: atlas
+    tag: v1.1.2
```
Comment (eddiepavkovic) [inline federation/values.yaml:34]: needs to be the latest atlas. 

```suggestion
    tag: v1.1.3
```

**Review by eddiepavkovic — CHANGES_REQUESTED** [review]

File: federation/templates/atlas_v1beta1_fedcluster.yaml (hunk 1 above)
Comment (bst525) [reply federation/templates/atlas_v1beta1_fedcluster.yaml:7]: Done.

File: federation/values.yaml (hunk 2 above)
Comment (bst525) [reply federation/values.yaml]: Checked iamsvc - value is 1.1.3 (no longer v1).  Made the change here to match.

**Review by SeemaTahaliyani — APPROVED** [review]

File: federation/values.yaml [hunk 3]
```diff
@@ -20,3 +20,22 @@ fedsvc:
...
+    pullPolicy: IfNotPresent
+    repository: atlas
+    tag: 1.1.3
+
+  # Persistent storage options for Atlas Cluster
+  persistence:
+    size: 10Gi
+
+  # Replica count for ObjectScale Federation Atlas Cluster
+  replicaCount: 3
```
Comment (fedor-chemashkin) [inline federation/values.yaml:41]: For small and micro profiles we use 1 atlas replica for ObjectStore

Comment (bst525) [reply federation/values.yaml:41]: @fedor-chemashkin Automation has been updated to use replicaCount=1.  this is same design used by IAM.

Comment (fedor-chemashkin) [reply federation/values.yaml:41]: From the user perspective, default charts installation will use 3 replicas even for small profiles

Comment (bst525) [reply federation/values.yaml]: @fedor-chemashkin I have updated it to 1

**Review by eddiepavkovic — APPROVED** [review]
ok

**Pull request merged**